### PR TITLE
Project proposal

### PR DESCRIPTION
**Pull request by listonjesse**
Adds a readme file which contains our project proposal

This proposal documents a simple summary of our product spec and goes over our project goals

### CODE REVIEW

**Comment by listonjesse** [comment]
Hey @XiuqinCandice, please feel free to leave comments and propose additions to this proposal.

Don't forget to click the preview button when reviewing the markdown file so you can see what it looks like when rendered using markdown:
image

This is a great opportunity to learn markdown if you don't already know. So here is a guide if you need it: https://guides.github.com/features/mastering-markdown/

File: ProjectProposal.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# What are we building
+We are building a simple web-app that allows users to maintain an online task list. 
+
+## Main features
+- Create, read, update, and delete tasks from the web interface
+- Tasks have a title and a description
+- Tasks can be marked as new, active, or completed
+- Tasks are stored in the cloud and can be accessed via a user account on any device
```
Comment (XiuqinCandice) [inline ProjectProposal.md]: Thank you @listonjesse This looks awesome, A question here: Is the '-' here is the same as '*' used in the markdown list?

File: ProjectProposal.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# What are we building
+We are building a simple web-app that allows users to maintain an online task list. 
+
+## Main features
+- Create, read, update, and delete tasks from the web interface
```
Comment (XiuqinCandice) [inline ProjectProposal.md]: Thank you @listonjesse . An initial idea here, shall we work with lists before work with tasks. 
'create, update, delete lists -> create, read, update tasks'

File: ProjectProposal.md (hunk 1 above)
Comment (XiuqinCandice) [reply ProjectProposal.md]: Q2: If I request changes, can I work on the same branch as you to help with the changes

File: ProjectProposal.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- Tasks can be marked as new, active, or completed
+- Tasks are stored in the cloud and can be accessed via a user account on any device
+
+## Stretch goals
+- Mobile interface
+- Due dates for tasks
+- Reminders
+- Organize tasks into groups and hierarchies
+- Mark certain tasks as important
+
```
Comment (XiuqinCandice) [inline ProjectProposal.md]: Very great job @listonjesse . Shall we add a search bar by keyword in the web app. Thank you so much for your work

File: ProjectProposal.md (hunk 1 above)
Comment (listonjesse) [reply ProjectProposal.md]: '-' and '*' both make a bullet in an unordered list

For question 2, I don't mind if you checkout this branch and push changes to it directly

File: ProjectProposal.md (hunk 3 above)
Comment (listonjesse) [reply ProjectProposal.md]: Feel free to add that to the stretch goals

File: ProjectProposal.md (hunk 2 above)
Comment (listonjesse) [reply ProjectProposal.md]: In my opinion, lists are covered in the stretch goals under "Organize tasks into groups and hierarchies". We can change the word from group to list if that makes more sense.

I think starting with all tasks on one page (just as if it were one list) will make the v1 implementation easier. We can add the ability to create multiple lists of tasks after we finish implementing tasks. It is usually easier to add structure and grouping to objects when the objects already exist.

With lists though, we should remember to add the ability to order tasks within a list. We can add that to the stretch goal

**Pull request merged**